src/components/feed/AddNode/AddNode.tsx

```diff
@@ -340,12 +354,18 @@ const AddNode: React.FC<AddNodeProps> = ({
 
   return (
     <React.Fragment>
+      {isCreator ? (
       <Button icon={<MdOutlineAddCircle />} type="button" onClick={toggleOpen}>
         Add a Child Node{" "}
         <span style={{ padding: "2px", color: "#F5F5DC", fontSize: "11px" }}>
           ( C )
         </span>
       </Button>
+      ) : (
+      <button disabled={true} >
```

Comment:
We have a sequence of buttons to render. Currently, they are:

- Add a Child Node (rw)
- Add a Pipeline (rw)
- Download Files (ro)
- View Terminal (ro)
- Add a Graph Node (rw)
- Delete Node (rw)

rw meaning "read-write" permissions needed, ro meaning "read-only" permissions. The feed owner has rw permissions, users who the feed has been shared with (but who do not own the feed) have ro permissions.

We should try to generalize the conditional that a button should be disabled and apply the solution to the other rw buttons.

Reply:
I'll try generalizing this, but I might need a little guidance.